### PR TITLE
Uncomment all event enum variants, add type alias and convert to derive

### PR DESCRIPTION
**Pull request by DevinR528**

### CODE REVIEW

File: src/room/canonical_alias.rs [hunk 1]
```diff
@@ -4,7 +4,12 @@ use ruma_events_macros::StateEventContent;
 use ruma_identifiers::RoomAliasId;
 use serde::{Deserialize, Serialize};
 
+use crate::StateEvent;
+
 /// Informs the room as to which alias is the canonical one.
+pub type CanonicalAliasEvent = StateEvent<CanonicalAliasEventContent>;
+
+/// The payload for `CanonicalAliasEvent`
```
Comment (jplatte) [inline src/room/canonical_alias.rs:12]: Just to keep consistent with the existing style, please add a dot at the end of each of these.

```suggestion
/// The payload for `CanonicalAliasEvent`.
```

File: src/room/encrypted.rs [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
-use crate::UnsignedData;
+use crate::StateEvent;
+
+/// An event that defines how messages sent in this room should be encrypted.
+pub type EncryptedEvent = StateEvent<EncryptedEventContent>;
 
 /// The payload for `EncryptedEvent`.
-#[derive(Clone, Debug, Deserialize, Serialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, StateEventContent)]
+#[ruma_event(type = "m.room.encrypted")]
```
Comment (jplatte) [inline src/room/encrypted.rs:17]: Please move this below `#[non_exhaustive]`, as we generally order builtin / `std` things first.

**Comment by jplatte** [comment]
Also, since this should be removing all remaining uses of `ruma_event!`, feel free to delete the macro itself too! :)

File: src/enums.rs [hunk 3]
```diff
@@ -38,20 +42,20 @@ event_enum! {
...
-        // "m.room.pinned_events",
-        // "m.room.power_levels",
-        // "m.room.server_acl",
-        // "m.room.third_party_invite",
-        // "m.room.tombstone",
-        // "m.room.topic",
+        "m.room.name",
+        "m.room.pinned_events",
+        "m.room.power_levels",
+        "m.room.server_acl",
```
Comment (jplatte) [inline src/enums.rs:55]: Could you add `m.room.redaction` too? It is conceptually a message event and has to be included in message / room collections, even though it has its own event type.

**Comment by jplatte** [comment]
Your branch has conflicts with the api22 though (probably because I removed room event support from `ruma_event!` yesterday). Can you rebase or merge that?

**Comment by DevinR528** [comment]
Yup already doing it ;)

**Pull request merged**